psi4/driver/procrouting/dft_funcs/superfuncs.py

```diff
@@ -108,6 +108,32 @@ def build_superfunctional(name, restricted):
     if core.has_option_changed("SCF", "DFT_ALPHA_C"):
         sup[0].set_c_alpha(core.get_option("SCF", "DFT_ALPHA_C"))
 
+    # customization of NL dispersion:
+    if (core.has_option_changed("SCF", "DFT_DISPERSION_PARAMETERS")):
+       nl_tuple = core.get_option("SCF", "DFT_DISPERSION_PARAMETERS")
+       sup[0].set_vv10_b(nl_tuple[0])
+       if len(nl_tuple) > 1:
+          sup[0].set_vv10_c(nl_tuple[1])
+       if len(nl_tuple) > 2:
+          raise ValidationError("too many entries in DFT_DISPERSION_PARAMETERS for DFT-NL")
+    # custom procedures using name 'scf' without any quadrature grid like HF will fail and are not detected
+    if (core.has_option_changed("SCF", "DFT_VV10_B") and core.has_option_changed("SCF", "DFT_VV10_C")):
+        if(name.lower()=='hf'):
+           raise ValidationError("SCF: HF with -NL not implemented")
+        vv10_b = core.get_option("SCF", "DFT_VV10_B")
+        vv10_c = core.get_option("SCF", "DFT_VV10_C")
+        sup[0].set_vv10_b(vv10_b)
+        sup[0].set_vv10_c(vv10_c)
+    elif core.has_option_changed("SCF", "DFT_VV10_B"):
+        if(name.lower()=='hf'):
+           raise ValidationError("SCF: HF with -NL not implemented")
+        vv10_b = core.get_option("SCF", "DFT_VV10_B")
+        sup[0].set_vv10_b(vv10_b)
+        if (abs(sup[0].vv10_c() - 0.0) <= 1e-8):
+            core.print_out("SCF: VV10_C not specified. Using default (C=0.0093)!")
+            sup[0].set_vv10_c(0.0093)
```

Comment:
It is for just provided vv10_b for an unknown functional and using the default C, which is the typical use case.

Different mechanisms are about user expectation. One can set omega manually, so also vv10_b/c should be possible. And since it behaves like -D in many ways, one would expect a similar functionality.

Reply:
Maybe call `DFT_DISPERSION_PARAMETERS` `VV10_DISPERSION_PARAMETERS` or `NL_`?

If someone sets `DFT_DISPERSION_PARAMETERS` and `DFT_VV10_B` it could be a bit confusing about order of operations, best to throw I think?